### PR TITLE
Updating Powershell SDK to include the HTTP APNS 2 changes

### PR DESCRIPTION
**Pull request by amolr**
## Description
1) Updating the powershell SDK to include the new HTTP 2 APNS changes
2) Updating the unit tests for the same

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/app.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/app.config]: @amolr please remove this file

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/app.config [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/app.config]: @amolr please remove this file

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj [hunk 3]
```diff
@@ -138,6 +138,9 @@
       <Link>AzureRM.NotificationHubs.psd1</Link>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="app.config">
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj]: @amolr please remove this reference

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/Commands.NotificationHubs.Test.csproj [hunk 4]
```diff
@@ -180,6 +180,9 @@
     <None Include="..\..\Common\Commands.ScenarioTests.ResourceManager.Common\AzureRM.Resources.ps1">
       <Link>ScenarioTests\AzureRM.Resources.ps1</Link>
     </None>
+    <None Include="app.config">
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/Commands.NotificationHubs.Test.csproj]: @amolr please remove this reference

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config [hunk 5]
```diff
@@ -4,7 +4,7 @@
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Gallery" version="2.6.2-preview" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.NotificationHubs" version="2.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.NotificationHubs" version="2.3.0-preview" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config]: use the new version

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config [hunk 6]
```diff
@@ -13,12 +13,12 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.22" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.5" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.5" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.7" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config]: use the newer versions

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config [hunk 7]
```diff
@@ -13,12 +13,12 @@
...
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.6" targetFramework="net452" />
   <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.2.9-preview" targetFramework="net45" />
   <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.5.0-preview" targetFramework="net45" />
   <package id="Microsoft.WindowsAzure.Management" version="4.1.1" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/packages.config]: do not use newtonsoft.json 9

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj [hunk 8]
```diff
@@ -51,15 +51,15 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.NotificationHubs, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.NotificationHubs.2.1.0-preview\lib\net45\Microsoft.Azure.Management.NotificationHubs.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.NotificationHubs.2.3.0-preview\lib\net452\Microsoft.Azure.Management.NotificationHubs.dll</HintPath>
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj]: Update the versions

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj [hunk 9]
```diff
@@ -74,8 +74,8 @@
       <HintPath>..\..\..\packages\Microsoft.Bcl.Async.1.0.168\lib\net40\Microsoft.Threading.Tasks.Extensions.Desktop.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands.NotificationHubs.csproj]: remove the version 9 depednency

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/packages.config [hunk 10]
```diff
@@ -3,12 +3,12 @@
...
   <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net45" />
   <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.5" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.5" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.7" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.6" targetFramework="net452" />
+  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/packages.config]: version 9

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/packages.config [hunk 11]
```diff
@@ -3,12 +3,12 @@
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.NotificationHubs" version="2.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.NotificationHubs" version="2.3.0-preview" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/packages.config]: Update the package and runtime versions

**Comment by markcowl** [comment]
on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1607/ 

signing run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1067/

LGTM when these pass

**Comment by shahabhijeet** [comment]
@amolr on-demand run failed due to wix.
Were you able to successfully create a msi locally.
Can you create a msi locally by executing tools\Installer\generate.ps1

**Comment by amolr** [comment]
Yes, the commit  	Adding Package 2.3.1 has the wix file generated locally

**Comment by amolr** [comment]
I tried to regenerate it using the command, but i dont see any changes to the wix file

**Comment by markcowl** [comment]
SO, it looks liek there are some files in the wxi you generated that are not in the build.  Can you:
* run ```git clean -xdf``` to remove all build products
* run ```ms build build.proj``` 
* rerun installer generation?

This should resolve the issue.

**Comment by markcowl** [comment]
on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1618/

signing run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1084/

**Pull request merged**